### PR TITLE
[SPARK-42106][PYTHON] Hide parameters when printing remote URL in REPL

### PR DESCRIPTION
**Pull request by nija-at**
### What changes were proposed in this pull request?

When the `pyspark` REPL is initialized with the `--remote <url>` option,
the URL provided is logged out to stdout as part of the REPL startup.
However, it may have auth tokens specified as query parameters.

This change is to print the network location, i.e., host and port, instead
of the full URL. Since Spark Connect servers will always be hosted at
the root and the URL scheme is always expected to be `sc`, only the
host and parts of the URL are relevant.

### Why are the changes needed?

Security best practices require not logging auth tokens to standard
output or any destination.

### Does this PR introduce _any_ user-facing change?

Previously -

```sh
$ ./bin/pyspark --remote "sc://foo.com/?x=y"
Python 3.10.8 (main, Nov 24 2022, 08:08:27) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /__ / .__/\_,_/_/ /_/\_\   version 3.4.0.dev0
      /_/

Using Python version 3.10.8 (main, Nov 24 2022 08:08:27)
Client connected to the Spark Connect server at sc://foo.com/?x=y
```

Now -

```sh
$ ./bin/pyspark --remote "sc://foo.com/?x=y"
Python 3.10.8 (main, Nov 24 2022, 08:08:27) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /__ / .__/\_,_/_/ /_/\_\   version 3.4.0.dev0
      /_/

Using Python version 3.10.8 (main, Nov 24 2022 08:08:27)
Client connected to the Spark Connect server at foo.com
```

```sh
$ ./bin/pyspark --remote "sc://foo.com:8080/?x=y"
Python 3.10.8 (main, Nov 24 2022, 08:08:27) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /__ / .__/\_,_/_/ /_/\_\   version 3.4.0.dev0
      /_/

Using Python version 3.10.8 (main, Nov 24 2022 08:08:27)
Client connected to the Spark Connect server at foo.com:8080
```

### How was this patch tested?

Manually tested by running the `pyspark` binary. See above.

### CODE REVIEW

File: python/pyspark/tests/test_url.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import unittest
+from pyspark.url import sanitize_url
+
+
+class UrlTests(unittest.TestCase):
+    def test_sanitize_url_valid(self):
```
Comment (grundprinzip) [inline python/pyspark/tests/test_url.py:22]: Can you have a look at `pyspark/sql/connect/client.py:ChannelBuilder` which has the parsing logic. The remote URLs are always encoded using `sc://....`. Not HTTP.

**Review by grundprinzip — COMMENTED** [review]
To enable the tests the, please add them to the according pyspark-connect module in `dev/sparktestsupport/modules.py`

File: python/pyspark/tests/test_url.py (hunk 1 above)
Comment (grundprinzip) [reply python/pyspark/tests/test_url.py:22]: Basically, make sure the scheme tests are testing for `sc://` as well :)

**Comment by HyukjinKwon** [comment]
I'll leave this to @grundprinzip .

**Review by srowen — APPROVED** [review]

File: python/pyspark/shell.py [hunk 2]
```diff
@@ -86,7 +88,7 @@
     % (platform.python_version(), platform.python_build()[0], platform.python_build()[1])
 )
 if is_remote():
-    print("Client connected to the Spark Connect server at %s" % (os.environ["SPARK_REMOTE"]))
+    print("Client connected to the Spark Connect server at %s" % urlparse(os.environ["SPARK_REMOTE"]).netloc)
```
Comment (srowen) [inline python/pyspark/shell.py:91]: I think this is fine; I don't know if security info would be in the URL (shouldn't be), and I wonder if some deployments want that info to clarify what's being connected to (like some kind of proxy), but can't think of a definite use case

**Review by grundprinzip — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**